### ISSUE TITLE
Generic.WhiteSpace.ScopeIndent false positive with nested `match`-es

### ISSUE THREAD

**Issue by andrey-yantsen** [PHPCSStandards/PHP_CodeSniffer#110]
## Describe the bug
The ScopeIndent sniff suggests a quite strange indentation when processing multi-nested `match`-statements

// It's a copy of my bug report from the old repo (squizlabs/PHP_CodeSniffer#3875)
// Thank you, Juliette, for an amazing initiative in keeping the project going!

### Code sample
```php
<?php

echo match (1) {
    0 => match (2) {
        2 => match (3) {
            3 => 3,
            default => -1,
        },
    },
    1 => match (2) {
        1 => match (3) {
            3 => 3,
            default => -1,
        },
        2 => match (3) {
            3 => 3,
            default => -1,
        },
    },
};
```

### How that code will look if formatted as suggested

```php
<?php

echo match (1) {
    0 => match (2) {
        2 => match (3) {
            3 => 3,
            default => -1,
        },
    },
    1 => match (2) {
        1 => match (3) {
            3 => 3,
            default => -1,
        },
            2 => match (3) {
                3 => 3,
                default => -1,
            },
    },
};
```

### Custom ruleset
N/A — reproducible with PSR12

### To reproduce
1. Create a file called `test.php` with the code sample above
2. Run `phpcs -s --standard=psr12 test.php`
3. See errors displayed
```
------------------------------------------------------------------------------------------------------------------------------
FOUND 4 ERRORS AFFECTING 4 LINES
------------------------------------------------------------------------------------------------------------------------------
 15 | ERROR | [x] Line indented incorrectly; expected at least 12 spaces, found 8
    |       |     (Generic.WhiteSpace.ScopeIndent.Incorrect)
 16 | ERROR | [x] Line indented incorrectly; expected at least 16 spaces, found 12
    |       |     (Generic.WhiteSpace.ScopeIndent.Incorrect)
 17 | ERROR | [x] Line indented incorrectly; expected at least 16 spaces, found 12
    |       |     (Generic.WhiteSpace.ScopeIndent.Incorrect)
 18 | ERROR | [x] Line indented incorrectly; expected 12 spaces, found 8 (Generic.WhiteSpace.ScopeIndent.IncorrectExact)
------------------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 4 MARKED SNIFF VIOLATIONS AUTOMATICALLY
------------------------------------------------------------------------------------------------------------------------------
```

## Expected behavior
No indentation errors.

## Versions (please complete the following information)

| | |
|-|-|
| Operating System | macOS 12.5
| PHP version | 8.0
| PHP_CodeSniffer version | 3.9.0
| Standard | PSR2
| Install type | Local composer

## Additional context
none

## Please confirm:

- [x] I have searched the issue list and am not opening a duplicate issue.
- [x] I confirm that this bug is a bug in PHP_CodeSniffer and not in one of the external standards.
- [x] I have verified the issue still exists in the `master` branch of PHP_CodeSniffer.

**Comment by ay-bmbl** [comment]
The issue is still present in v3.9.0.

**Comment by andrey-yantsen** [comment]
The issues is still present in v3.9.0.

**Comment by andrey-yantsen** [comment]
@jrfnl I'm in no way rushing it, and I want to do a patch at some later date. Sorry if it sounded like an attack, that wasn't my intention :)

I just saw that there were some changes around the processing of `match`-es in the last releases, and I wanted to save some time on retesting to anybody stumbling upon this ticket.

**Comment by jrfnl** [comment]
Pff.. this was complex one. PR #502 should fix it though. Testing appreciated.

**Comment by andrey-yantsen** [comment]
Thank you, @jrfnl!

I just checked the master branch — the issue is gone for all the examples I had at hand.

**Comment by jrfnl** [comment]
Thanks @andrey-yantsen for testing and confirming!

**Comment by mdeboer** [comment]
Can we re-open this? I can confirm that this ticket solved a lot of issues but I found another scenario where this isn't fixed yet.

In case where a _method_ has attributes, formatted like this, the false positive would still occur. Note that this only occurs when a method has attributes and only when exactly formatted like this. Attributes on the class have no effect.

```php
<?php

class Foo
{
    #[
        AttributeA(),
        AttributeB()
    ]
    public function foo(): int
    {
        $bar = 'bar';

        $result = match ($bar) {
            'foo' => true,
            'bar' => 300,
            'baz' => 'hello world',
            default => null
        };

        if ($result === null) {
            return 404;
        }

        return 200;
    }
}
```

**Output**
```diff
--- test.php
+++ PHP_CodeSniffer
@@ -17,10 +17,10 @@
             default => null
         };
 
-        if ($result === null) {
-            return 404;
-        }
+            if ($result === null) {
+                return 404;
+            }
 
-        return 200;
+            return 200;
     }
 }
```

When formatted like this either with one or multiple attributes, everything is ok and the output is empty.

```php
<?php

class Foo
{
    #[AttributeA()]
    #[AttributeB()]
    public function foo(): int
    {
        $bar = 'bar';

        $result = match ($bar) {
            'foo' => true,
            'bar' => 300,
            'baz' => 'hello world',
            default => null
        };

        if ($result === null) {
            return 404;
        }

        return 200;
    }
}
```

**Comment by jrfnl** [comment]
@mdeboer Please open a new issue.
